src/base/QXmppJingleIq.cpp

```diff
@@ -1462,3 +1572,366 @@ bool QXmppJinglePayloadType::operator==(const QXmppJinglePayloadType &other) con
+{
+    d->subtype = subtype;
+}
+
+///
+/// Returns the parameters for RTP Feedback Negotiation.
+///
+/// \return the RTP Feedback Negotiation parameters
+///
+QVector<QXmppSdpParameter> QXmppJingleRtpFeedbackTypeAndParameters::parameters() const
+{
+    return d->parameters;
+}
+
+///
+/// Sets the parameters for RTP Feedback Negotiation.
+///
+/// Additional parameters in the form "a=b" can be set by this method.
+/// Any other form of parameters can be provided by setting the name and a default-constructed
+/// QString as the value of QXmppJingleRtpFeedbackTypeAndParameters::Parameter.
+/// If there is only one parameter, use QXmppJingleRtpFeedbackTypeAndParameters::setSubtype()
+/// instead of this method.
+///
+/// \param parameters RTP Feedback Negotiation parameters
+///
+void QXmppJingleRtpFeedbackTypeAndParameters::setParameters(const QVector<QXmppSdpParameter> &parameters)
+{
+    d->parameters = parameters;
+}
+
+/// \cond
+void QXmppJingleRtpFeedbackTypeAndParameters::parse(const QDomElement &element)
+{
+    d->type = element.attribute(QStringLiteral("type"));
+    d->subtype = element.attribute(QStringLiteral("subtype"));
+
+    QVector<QXmppSdpParameter> parameters;
+    for (auto childElement = element.firstChildElement();
+         !childElement.isNull();
+         childElement = childElement.nextSiblingElement()) {
+        if (QXmppSdpParameter::isSdpParameter(childElement)) {
+            QXmppSdpParameter parameter;
+            parameter.parse(childElement);
+            parameters.append(parameter);
+        }
+    }
+    d->parameters = parameters;
+}
+
+void QXmppJingleRtpFeedbackTypeAndParameters::toXml(QXmlStreamWriter *writer) const
+{
+    writer->writeStartElement(QStringLiteral("rtcp-fb"));
+    writer->writeDefaultNamespace(ns_jingle_rtp_feedback_negotiation);
+    helperToXmlAddAttribute(writer, QStringLiteral("type"), d->type);
+
+    // If there are parameters, they must be used instead of the subtype.
+    if (const auto subtype = d->subtype; subtype.isEmpty()) {
+        for (const auto &parameter : std::as_const(d->parameters)) {
+            parameter.toXml(writer);
+        }
+    } else {
+        helperToXmlAddAttribute(writer, QStringLiteral("subtype"), subtype);
+    }
+
+    writer->writeEndElement();
+}
+/// \endcond
+
+///
+/// Determines whether the given DOM element is an RTP Feedback Negotiation type and parameters
+/// element.
+///
+/// \param element DOM element being checked
+///
+/// \return whether element is an RTP Feedback Negotiation type and parameters element
+///
+bool QXmppJingleRtpFeedbackTypeAndParameters::isJingleRtpFeedbackTypeAndParameters(const QDomElement &element)
+{
+    return element.tagName() == QStringLiteral("rtcp-fb") &&
+        element.namespaceURI() == ns_jingle_rtp_feedback_negotiation;
+}
+
+class QXmppJingleRtpFeedbackIntervalPrivate : public QSharedData
+{
+public:
+    uint32_t value;
+};
```

Comment:
this is so small, it shouldn't have a d-ptr and please use uint64_t (so we later have the option to add a d ptr)